docs/developer/troubleshooting.rst

```diff
@@ -195,4 +195,25 @@ funcion for example) as it is shown in the example below:
   pd_df = pd_df.set_index(index_col_name)
   pd_df.index.name = None
 
+Error when using OmniSci engine: ``LLVM ERROR: inconsistency in registered CommandLine options``
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+This can happen when you set the following config values prior Modin import:
+
+.. code-block:: python
+
+  import modin.config as cfg
+  cfg.Engine.put("Native")  # 'omniscidbe'/'dbe' would be imported with dlopen flags first time
+  cfg.StorageFormat.put("Omnisci")
+  cfg.IsExperimental.put(True)
+  import modin.pandas as pd  # Error
```

Comment:
This should not be a part of docs, this is a bug. It should not happen, as setting variables like this is perfectly normal - this is what we recommend doing for a lot of other cases (and I think "Using OmniSci" page recommended doing exactly this a while ago)!

That said, I think that with latest code changes, this is no longer a case anyway. I propose to change this part to say "do not import OmniSci directly, use `PyDbEngine` from `modin.experimental.core.execution.native.implementations.omnisci_on_native.utils`" or something similar.

Reply:
Right now we recommend setting just cfg.StorageFormat.put("Omnisci") which should work as far as I understand

Reply:
>This should not be a part of docs, this is a bug. It should not happen, as setting variables like this is perfectly normal - this is what we recommend doing for a lot of other cases (and I think "Using OmniSci" page recommended doing exactly this a while ago)!

In my understanding, the problem described here is not related to using proper dlopen flags but refers to multiple LLVM libraries usage problem. And this problem still exists. We cannot force other libraries to use the same LLVM library and link it dynamically. So I wouldn't say we have a bug here. But there is still a problem we try to propose a workaround for.

Reply:
Yeah, we found that importing `pyarrow.gandiva` makes this segfault happen.
**Yaroslav** would be making an updated version of this PR which makes this usecase work soon.

Meanwhile someone should address LLVM incompatibility of OmniSci and Gandiva (sometime in the future)...

Reply:
fixed